pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -137,6 +144,53 @@ func ensureOpenshiftIngress(instance *servingv1alpha1.KnativeServing) error {
 	return nil
 }
 
+func installServiceMonitor(instance *servingv1alpha1.KnativeServing) error {
+	namespace := instance.GetNamespace()
+	log.Info("Installing ServiceMonitor")
+	const path = "deploy/service_monitor.yaml"
+
+	if serviceMonitorExists, err := serviceMonitorExists(namespace); err != nil {
+		return err
+	} else if !serviceMonitorExists {
+		log.Info("ServiceMonitor CRD is not installed. Skip to install ServiceMonitor")
+		return nil
+	}
+
+	// Add label openshift.io/cluster-monitoring to namespace
+	ns := &corev1.Namespace{}
+	if err := api.Get(context.TODO(), client.ObjectKey{Name: namespace}, ns); err != nil {
+		if !meta.IsNoMatchError(err) {
```

Comment:
Why do we ignore this error?

Reply:
Thank you. It should not be ignored as it is unusual that the namespace is not found. Updated.